src/tabs/map.rs

```diff
@@ -55,6 +57,29 @@ pub struct Tab {
     /// When drawing with any brush,
     /// this is set to the position of the original tile we began drawing on
     drawing_shape_pos: Option<egui::Pos2>,
+
+    /// Undo history
+    history: VecDeque<HistoryEntry>,
+    /// When operations are undone, they are put here so that they can be redone
+    redo_history: Vec<HistoryEntry>,
+    /// This is the buffer for the tilemap states for the undo history
+    tilemap_history: Table3,
```

Comment:
This seems a little wasteful. Would it be better to store a delta of changed tiles in a `HistoryEntry`?

Reply:
Yeah I suppose you're right. I'm in class right now so give me a moment.